idearium-lib/common/mq/publisher.js

```diff
@@ -1,26 +1,23 @@
 'use strict';
 
 const manager = require('./messages');
-const { hasProperty } = require('../../lib/util');
 const log = require('../log')('idearium-lib:common:mq/publisher');
 
 /**
  * Check if a RabbitMQ message exists before publishing.
  * @param {String} type Message type.
  * @param {Object} data Message data.
- * @return {Void} Publishes the message.
+ * @return {Promise} Publishes the message.
  */
 const publish = (type, data) => {
 
-    if (!hasProperty(manager.messages, type)) {
-        return log.error(`Message of type: ${type} not found`);
-    }
+    return manager.publish(type, data)
```

Comment:
Is this code missing a promise? Also, I think it can be simplified too since the new `consume` and `publish` functions already print these messages.
```javascript
const publish = (type, data) => manager.publish(type, data);
```

Reply:
In fact do we even need this anymore?

Reply:
@allanchau, Strictly, we don't need it. However, it automatically applies the `log.debug` so I think it's still worthwhile and useful. What do you think?

Reply:
But you have removed it in this pr?

Reply:
What do you mean removed it? It's still there?

Reply:
oh I meant `return log.error(`Message of type: ${type} not found`);`
I thought this had already been merged https://github.com/idearium/idearium-lib/pull/26

Reply:
that contains the auto log
```javascript
debug('Publishing message: %O', {
                data,
                exchange,
                routingKey,
            });
```

Reply:
That's been moved into an improved `mq.Manager().publish` function, which this function calls. That function doesn't log successful publishing though, which is why I thought we should keep this common file.

Reply:
I see, but that is still a work in progress. We can remove this when we get back to that PR. But until then, I think it's useful.

Reply:
ok sounds good.